ADR/ADR004-store-form-schemas-in-an-api.md

```diff
@@ -0,0 +1,31 @@
+# ADR004: Store form schemas in an API
+
+Date: 2022-02-18
+
+## Status
+
+Proposed
+
+## Context
+
+- Manager/Designer components don't need to be as scalable as the runner
+- Runner needs to scale to number of users
+  - Sustainability, runner not needed during quiet times
+- API needs to scale
+
+## Decision
+
+```mermaid
+graph LR
+  Manager[Form manager]
+  API[Forms API]
+  Runner[Forms runner]
+  Designer[Form designer]
+  Manager --> API
+  Designer --> API
+  API --> Runner
+  Runner --> API
+```
+## Consequences
+
+> both positive and negative consequences of the decision
```

Comment:
I think the concequences are both scalability and cost.

We can ramp up the public side of the service, the runner and api, to meet demand, and then scale it back dramatically (1000x maybe, even) to save running costs.

Reply:
Yeah trade off and costs are probably one of the big things to consider - I'm not sure the cost/scalability trade offs of say, the designer/manager portion vs the runner - wonder if one ends up being cheaper to run if it was separate vs as a monolith